aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -367,22 +370,88 @@ public void onResult(UserStateDetails result) {
 
                 @Override
                 public void onError(Exception exception) {
-                    onError.accept(new AuthException(
+                    onException.accept(new AuthException(
                             "An error occurred while attempting to retrieve your user details",
                             exception,
                             "See attached exception for more details"
                     ));
                 }
             });
         } catch (Throwable exception) {
-            onError.accept(new AuthException(
+            onException.accept(new AuthException(
                     "An error occurred fetching authorization details for the current user",
                     exception,
                     "See attached exception for more details"
             ));
         }
     }
 
+    @Override
+    public void forgotPassword(
+            @NonNull String username,
+            @NonNull Consumer<AuthCodeDeliveryDetails> onSuccess,
+            @NonNull Consumer<AuthException> onException
+    ) {
+        AWSMobileClient.getInstance().forgotPassword(username, new Callback<ForgotPasswordResult>() {
+            @Override
+            public void onResult(ForgotPasswordResult result) {
+                if (result.getState().equals(ForgotPasswordState.CONFIRMATION_CODE)) {
+                    onSuccess.accept(convertCodeDeliveryDetails(result.getParameters()));
+                } else {
+                    onException.accept(new AuthException(
+                            "Received an unsupported response after triggering password recovery: " + result.getState(),
+                            "This is almost certainly a bug. Please report it as an issue in our GitHub repo."
+                    ));
+                }
+            }
+
+            @Override
+            public void onError(Exception exception) {
+                onException.accept(new AuthException(
+                        "An error occurred triggering password recovery",
+                        exception,
+                        "See attached exception for more details"
+                ));
+            }
+        });
+    }
+
+    @Override
+    public void confirmForgotPassword(
+            @NonNull String newPassword,
+            @NonNull String confirmationCode,
+            @NonNull Action onSuccess,
+            @NonNull Consumer<AuthException> onException
+    ) {
+        AWSMobileClient.getInstance().confirmForgotPassword(
+            newPassword,
+            confirmationCode,
+            new Callback<ForgotPasswordResult>() {
+                @Override
+                public void onResult(ForgotPasswordResult result) {
+                    if (result.getState().equals(ForgotPasswordState.DONE)) {
+                        onSuccess.call();
```

Comment:
How come you don't pass back `convertCodeDeliveryDetails(result.getParameters())` here too? Doesn't contain any valuable info?

Reply:
Yeah no it's a messed up implementation in AWSMobileClient where there's only one legitimate value which is Done and everything else is empty or an error. :P

Reply:
Haha oh alright!